### PR TITLE
PWX-37682 | Add missing field for StorageCluster Spec in helm

### PR DESCRIPTION
**Pull request by hitesh-wani-px**
**What this PR does / why we need it**:
This will following fields in the storageClusterSpec
```yaml
runtimeOptions:       map[string]string              
featureGates:     map[string]string                  
metadata:
  annotation: object
  labels: object
security: 
  enabled: bool                    
  auth:
    guestAccess: string            
    selfSigned:                      
      tokenLifetime:  string
      issuer: string                     
      sharedSecret:  string
```

**Which issue(s) this PR fixes** (optional)
Closes #https://purestorage.atlassian.net/browse/PWX-37682

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by hitesh-wani-px** [comment]
**Testing Notes:**
1. Add following values in `values.yaml`
```yaml
runtimeOptions:  num_io_threads=10
featureGates:  CSI=true
security: 
  enabled: true                    
  auth:
    guestAccess: Enabled             
    selfSigned:                      
      tokenLifetime: "24h"                
      issuer:    "my-issuer"                  
      sharedSecret: "px-shared-secret"
```
2. Install PX
```
helm install my-release --set internalKVDB=true,clusterName=$(uuidgen) ./charts/portworx/ -n portworx --debug
```
3. The generated storageCluster template should incorporate all specified values.
```yaml
kind: StorageCluster
apiVersion: core.libopenstorage.org/v1
metadata:
  name: "mycluster"
  namespace: kube-system
  annotations:
  labels:
    heritage: "Helm"
    release: "my-release"
    chart: "portworx-3.1.0"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/instance: "my-release"
spec:
  image: portworx/oci-monitor:3.1.0
  imagePullPolicy: Always

  kvdb:
    internal: true
  storage:
    useAll: true
  secretsProvider: k8s

  env:
  - name: PX_SECRETS_NAMESPACE
    value: portworx

  stork:
    enabled: true
  csi:
    enabled: false
  runtimeOptions:
    num_io_threads: "10"
  featureGates:
    CSI: "true"
  security: 
    enabled: true
    auth:
      guestAccess: Enabled
      selfSigned:
        tokenLifetime: 24h
        issuer: my-issuer
        sharedSecret: px-shared-secret
```
3. We should be able to see the following configurations reflected on the cluster:
    1. `num_io_threds=10` should be pass as arguments in PX pod
    ```yaml
        containers:
        - args:
        - -c
        - mycluster
        - -x
        - kubernetes
        - -b
        - -a
        - -secret_type
        - k8s
        - -rt_opts
        - num_io_threads=10
    ```
    2. With the feature gate CSI enabled, we should be able to see the CSI pods running in the cluster.
    3. `px-shared-secret` should be created in portworx installed namespace.
         ```
         kubectl get secrets -n portworx
         ```
    4. We should able to create and use secure pvc for mysql deployment
    ```yaml
    kind: PersistentVolumeClaim
    apiVersion: v1
    metadata:
      name: px-mysql-pvc
      annotations:
        openstorage.io/auth-secret-name: px-shared-secret
    spec:
      storageClassName: px-csi-db
      accessModes:
        - ReadWriteMany
      resources:
        requests:
          storage: 2Gi
    ```
    ```yaml
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      name: mysql
    spec:
      selector:
        matchLabels:
          app: mysql
      strategy:
        rollingUpdate:
          maxSurge: 1
          maxUnavailable: 1
        type: RollingUpdate
      replicas: 1
      template:
        metadata:
          labels:
            app: mysql
            version: "1"
        spec:
          containers:
          - image: mysql:5.6
            name: mysql
            env:
            - name: MYSQL_ROOT_PASSWORD
              value: password
            ports:
            - containerPort: 3306
            volumeMounts:
            - name: mysql-persistent-storage
              mountPath: /var/lib/mysql
          volumes:
          - name: mysql-persistent-storage
            persistentVolumeClaim:
              claimName: px-mysql-pvc
     ```
     ```
    hitesh@hitesh-wani:~$ kubectl get pod -n portworx | grep mysql
    mysql-549984cbcf-f99xh                      1/1     Running   0             14s
     ```

**Comment by nikita-bhatia** [comment]
@hitesh-wani-px did we miss "resources" and "metadata" here?

**Comment by hitesh-wani-px** [comment]
> @hitesh-wani-px did we miss "resources" and "metadata" here?

Yes, @nikita-bhatia, I missed these, but I have now added these fields.

File: test/portworx/storagecluster_helm_template_test.go [hunk 1]
```diff
@@ -24,39 +24,87 @@ func TestStorageClusterHelmTemplate(t *testing.T) {
 		helmOption     *helm.Options
 		resultFileName string
 	}{
+		// {
```
Comment (nikita-bhatia) [inline test/portworx/storagecluster_helm_template_test.go:27]: why is following tests commented?

Comment (hitesh-wani-px) [reply test/portworx/storagecluster_helm_template_test.go]: Removed
commented it by mistake while running tests.

**Review by nikita-bhatia — APPROVED** [review]
LGTM

**Pull request merged**